snappy_wrappers/wrappers/somatic_variants_checking/summarize-vcf.py

```diff
@@ -0,0 +1,282 @@
+# import packages
+import argparse
+import gzip
+import json  # for writing out the json file
+
+from cyvcf2 import VCF  # for reading vcf file
+import tabix
+
+
+def check_variant_in_bed(v_chrom, v_start, v_end, bed_intervals, padding=0):
+    try:
+        tmp = []
+        records = bed_intervals.query(v_chrom, v_start - int(padding), v_end + int(padding))
+        for record in records:
+            tmp.append(record)
+        if len(tmp) >= 1:
+            return True
+        else:
+            return False
+    except OSError:
+        print("You should index the bed file with tabix first!")
+
+
+def get_contigs_from_bed_file(bedfile):
+    contigs = []
+    try:
+        with gzip.open(bedfile, "r") as intervals:
+            for interval in intervals:
+                line = interval.decode()
+                tmp = line.split("\t")[0]
+                if tmp not in contigs:
+                    contigs.append(tmp)
+        return contigs
+    except OSError:
+        print("Error reading the GZIP file.")
+
+
+def get_variant_type(ref, alt):
+    if len(ref) == len(alt):
+        if len(alt) == 1:
+            return "SNV"
+        else:
+            return "ONV"
+    elif len(alt) < len(ref):
+        return "indel"
+
+
+def check_sp_read(variant, pos_sample, minimal, limited):
+    dp = variant.format("AD")[1][pos_sample]
+    if dp <= minimal:
+        return "minimal"
+    elif (dp > minimal) and (dp <= limited):
+        return "limited"
+    elif dp >= limited:
+        return "strong"
+
+
+def assign_class_snvs(variant, mt_mat):
+    temp = str(variant.REF) + ">" + str(variant.ALT[0])
+    # Following result from plot-vcfstats with bcftools
+    if temp in ["A>C", "T>G"]:
+        mt_mat[0] += 1
+    elif temp in ["A>G", "T>C"]:
+        mt_mat[1] += 1
+    elif temp in ["A>T", "T>A"]:
+        mt_mat[2] += 1
+    elif temp in ["C>G", "G>C"]:
+        mt_mat[3] += 1
+    elif temp in ["C>T", "G>A"]:
+        mt_mat[4] += 1
+    elif temp in ["G>T", "C>A"]:
+        mt_mat[5] += 1
+    return mt_mat
```

Comment:
I am not sure it is good practice to return the list (see [this example](https://stackoverflow.com/questions/26027694/correct-style-for-python-functions-that-mutate-the-argument))